src/AddImageChecks.cpp

```diff
@@ -478,8 +479,23 @@ Stmt add_image_checks(Stmt s,
             // Check the var passed in equals the constrained version (when not in inference mode)
             asserts_constrained.push_back(AssertStmt::make(var == constrained_var, error));
         }
+        if (param.defined() && param.host_alignment() > 1) {
+            string host_name = name + ".host";
+            int alignment_required = param.host_alignment();
+            Expr host_ptr = Variable::make(Int(32), host_name);
```

Comment:
Actually, I think it should be `(cast<uint64_t>(host_ptr) % alignment_required) == 0`. Alignment that isn't a power of 2 seems crazy, but there's no reason not to do it this way anyways.